srv/components/misc_pkgs/kafka/stop.sls

```diff
@@ -25,24 +25,38 @@ Stop kafka:
   service.dead:
     - name: kafka
 
-#TODO: find better solution to add delay
-Wait for kafka to stop:
-  module.run:
-    - test.sleep:
-      - length: 10
+Ensure kafka has stopped:
+  cmd.run:
+    - name: test 1 -le $(ps ax | grep -i 'kafka.Kafka' | grep -v grep | awk '{print $1}' | wc -l)
+    - retry:
+    # Ref: https://docs.saltproject.io/en/3000/ref/states/requisites.html#retrying-states
+        attempts: 10
+        until: False
+        interval: 2
     - require:
-      - Stop kafka
+      - Start kafka
 
 #Stop zookeeper:
 #  cmd.run:
 #    - name: ./bin/zookeeper-server-stop.sh -daemon config/zookeeper.properties
 #    - cwd: /opt/kafka
 #    - onlyif: test 1 -le $(ps ax | grep java | grep -i QuorumPeerMain | grep -v grep | awk '{print $1}' | wc -l)
 #    - require:
-#      - Wait for kafka to stop
+#      - Ensure kafka has stopped
 
 Stop zookeeper:
   service.dead:
     - name: kafka-zookeeper
     - require:
-      - Wait for kafka to stop
+      - Ensure kafka has stopped
+
+Ensure kafka-zookeeper has stopped:
+  cmd.run:
+    - name: test 1 -le $(ps ax | grep java | grep -i QuorumPeerMain | grep -v grep | awk '{print $1}' | wc -l)
```

Comment:
Use service command

Reply:
@sachinpunadikar the section on line 47 takes care of this requirement.
```
Stop zookeeper:
  service.dead:
    - name: kafka-zookeeper
    - require:
      - Ensure kafka has stopped
 ```
Salt ensures that service has actually started in this block.